### ISSUE TITLE
Scan doesn't find variations with special characters

### ISSUE THREAD

**Issue by jacobsteffler** [marcstraube/foundryvtt-scenery#17]
**Describe the bug**
Scanning is not able to identify variations when the common part of the filename includes characters that will be URI-encoded.
This includes spaces, symbols, and non-English characters.

**To Reproduce**
1. Create two map files where one basename contains the other, but the name includes spaces.
For example, "Forest Path.jpg" and "Forest Path Night.jpg".
2. Create a new scene using Forest Path as the background image.
3. Open the Scenery popup and scan for variations. Nothing is found.

**Expected behavior**
"Forest Path Night" should be identified as a variant.

**System information (please complete the following information):**
 - Scenery Version: 1.1.3
 - FoundryVTT Version: 11.315
 - Browser: Firefox
 - FoundryVTT Game System and Version: dnd5e 2.4.1

**Additional context**
FilePicker API returns URI-encoded paths. I will submit a PR to use decodeURI in the scan process.
Arguably it is good practice to use filenames that don't get encoded, but Foundry in general handles special characters without issue.

**Comment by marcstraube** [comment]
Hi @jacobsteffler,

sorry for taking some time to respond to your issue.

I've tried to reproduce the behavior as described (FoundryVTT v11 and v12; native Client and Firefox), but was not able to. I scanned for images with spaces in the file and path name. At first without your fix, than with your fix #18 .

In both cases the variant file was found! The only difference without your fix was, that the space character was replaced with the decoded **%20** entity.

I've added another file named **Forest Path Night & Rain.jpg** and that was also found in both cases. And in both cases the ampersand was replaced with its decoded entity **%26**.

Do you have another example for me?

**Comment by jacobsteffler** [comment]
Hi, no worries.

I just tested this again with Foundry 12.331 and Scenery 1.2.0 (without my patch).
This seems to be working fine for me too now.

I remember when I first saw this, the Scenery popup showed the default file path with the original characters (space and &).
But this time, it shows the encoded characters, %20 and %26, from the start.
With those encoded characters in place, the search algorithm finds the variation files.

Thanks for looking into it, closing this issue.
